docs/api/console.md

```diff
@@ -0,0 +1,159 @@
+# 🎮 Console
+
+An implementation of the standard `console` API.
+
+## Quick start
+
+### Log messages
+
+`Console` provides the following general purpose logging functions.
+* `console.log` for general messages.
+* `console.info` for information messages.
+* `console.warn` for warnings.
+* `console.error` for errors.
+* `console.debug` for debug level messages.
+
+Each of these has a similar interface. Each function accepts an arbitrary number of arguments. The logged message consists of a human readable representation of the arguments, separated by strings.
+
+The following two lines will both log the message "Hello World".
+```typescript
+console.log("Hello World!");
+console.log("Hello", "World!");
+```
+
+In practice using multiple arguments is unwieldy and difficult to read.
+Most javascript guildelines advise using string interpolation instead of multiple arguments.
+
+For example the following
+``` typescript
+function sayHello(name: string) {
+  console.log(`Hello ${name} from jstz!`);
+}
+```
+should be preferred to
+
+``` typescript
+function sayHello(name: string) {
+  console.log("Hello", name, "from jstz!");
+}
+```
+
+Different logging functions can be combined to make logs more informative and to allow easy filtering.
+
+```typescript
+function myAwesomeFunction(arg: unknown) {
+    console.debug(`Entered myAwesomeFunction with argument ${arg}`)
+    if (typeof arg !== 'string') {
+    console.warn("Converting non string argument to string");
+    arg = arg.toString();
+    }
+    try {
+        let result = functionThatMayThrow(arg);
+        console.info(`Result: ${result}`);
+    } catch (error) {
+        console.error(`Call failed with message ${error}`);
+        throw error;
+    }
+}
+```
+### Assertions
+
+`console.assert` will log an error message if its first argument is falsy.
+It will not throw an error or interrupt control flow, but can be used to flag places where 
+something unexpected occurs.
+If the first argument is truthy there will be no effect.
```

Comment:
I think this could be simpler:
```suggestion
`console.assert` method logs an error message if the assertion is false. If the assertion is true, nothing happens.
```